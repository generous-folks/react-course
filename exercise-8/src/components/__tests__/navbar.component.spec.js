import React from 'react';

import { shallow } from 'enzyme';

import { AppBar, IconButton, Menu, Typography } from '@material-ui/core';

import NavBar from '../navbar.component';

let wrapper;
jest.mock('../../modules/user/user.context.js', () => ({
  useUserState: jest.fn().mockReturnValue({ user: null }),
  useUser: jest.fn().mockReturnValue([{ user: null }, jest.fn()]),
}));
const getWrapper = () => shallow(<NavBar />);

describe('<NavBar />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = getWrapper();
  });
  describe('Snapshot', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Markup checks', () => {
    it('should contain the correct markup', () => {
      expect(wrapper.find(AppBar).exists()).toBeTruthy();
      expect(
        wrapper
          .find(Typography)
          .first()
          .text(),
      ).toBe('Shopping App');
    });
  });

  describe('Interaction checks', () => {
    it('should sur anchorEl from clicked item', () => {
      expect(wrapper.find(Menu).prop('open')).toBeFalsy();
      wrapper
        .find(IconButton)
        .last()
        .prop('onClick')({ currentTarget: 'foo' });
      wrapper.update();
      expect(wrapper.find(Menu).prop('open')).toBeTruthy();
    });
  });
});
