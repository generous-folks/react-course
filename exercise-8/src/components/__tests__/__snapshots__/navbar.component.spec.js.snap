// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> Snapshot should render correctly 1`] = `
<WithStyles(ForwardRef(AppBar))
  position="static"
>
  <WithStyles(ForwardRef(Toolbar))>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-5"
      variant="h6"
    >
      Shopping App
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(IconButton))
      aria-label="logout button"
      className="makeStyles-menuButton-2 makeStyles-logoutButton-4"
      edge="start"
      onClick={[Function]}
    >
      <ForwardRef(PowerSettingsNewOutlinedIcon) />
    </WithStyles(ForwardRef(IconButton))>
    <div>
      <WithStyles(ForwardRef(IconButton))
        aria-controls="menu-appbar"
        aria-haspopup="true"
        aria-label="menu"
        className="makeStyles-menuButton-2"
        color="inherit"
        edge="start"
        onClick={[Function]}
      >
        <ForwardRef(MenuIcon) />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Menu))
        anchorEl={null}
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        id="menu-appbar"
        keepMounted={true}
        onClose={[Function]}
        open={false}
        transformOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
      >
        <WithStyles(ForwardRef(MenuItem))
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          onClick={[Function]}
          to="/"
        >
          Home
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          onClick={[Function]}
          to="/contact"
        >
          Contact
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem))
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          onClick={[Function]}
          to="/about"
        >
          About
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(Menu))>
    </div>
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;
